static/scripts/rewards/render-transaction/utils.ts

```diff
@@ -0,0 +1,5 @@
+export function removeAllEventListeners(element: Element): Element {
```

Comment:
You must remove the event listener manually. Deleting an element with an event listener from the DOM is not sufficient. I verified this with a quick perplexity search.

Reply:
The problem is the mintNftHandler returns a new function every time but if you want to remove an event listener you need to give it the function that mintNftHandler returned when it was registered. It's not enough to call mintNftHandler again with the same parameters because it returns a new function so it doesn't work.

I'm pretty sure cloning an element and replacing it with the old one removes all event listeners because I tested it. You can also see [here](https://developer.mozilla.org/en-US/docs/Web/API/Node/cloneNode).

Reply:
There's nothing on that page that says the event listener is removed when an element is removed from the DOM.